src/i2c/src/main.rs

```diff
@@ -226,41 +227,48 @@ mod tests {
         }
     }
 
-    fn get_cmd_args(name: &str, devices: &str, count: u32) -> ArgMatches {
+    fn get_cmd_args(devices: &str, count: Option<u32>) -> ArgMatches {
         let yaml = load_yaml!("cli.yaml");
         let app = App::from(yaml);
 
-        if count != 0 {
+        if let Some(count) = count {
```

Comment:
It would be better to have the vec definition in the if, and call `app.try_get_matches_from` after the if to make the code more clear.

We could write it as follows:

```rust
        let mut args = vec!["prog", "-s", SOCKET_NAME, "-l", devices];

        let socket_count_str;
        if let Some(socket_count) = count {
            socket_count_str = socket_count.to_string();
            args.extend_from_slice(&["-c", &socket_count_str]);
        }
        app.try_get_matches_from(args).unwrap()
```